### PR TITLE
Support success policy for MXJob

### PR DESCRIPTION
**Pull request by jasonliu747**
successPolicy would determine the success of the job, user configurable.

### CODE REVIEW

File: manifests/base/deployment.yaml [hunk 1]
```diff
@@ -17,7 +17,7 @@ spec:
       serviceAccountName: mxnet-operator
       containers:
       - name: mxnet-operator
-        image: mxjob/mxnet-operator:v1beta1
+        image: jasonliu747/mxnet-operator:v1beta1
```
Comment (gaocegege) [inline manifests/base/deployment.yaml:20]: Please do not change the image here

Comment (jasonliu747) [reply manifests/base/deployment.yaml:20]: :ok_hand:

File: pkg/apis/mxnet/v1beta1/types.go [hunk 2]
```diff
@@ -45,9 +45,13 @@ type MXJob struct {
 type MXJobSpec struct {
 	// CleanPodPolicy defines the policy to kill pods after MXJob is
 	// succeeded.
-	// Default to Running.
+	// Default to All.
```
Comment (terrytangyuan) [inline pkg/apis/mxnet/v1beta1/types.go:48]: Why are we changing the default here?

File: pkg/apis/mxnet/v1beta1/types.go [hunk 3]
```diff
@@ -45,9 +45,13 @@ type MXJob struct {
...
 	CleanPodPolicy *CleanPodPolicy `json:"cleanPodPolicy,omitempty"`
 
+	// SuccessPolicy defines the policy to mark the MXJob as succeeded.
+	// Default to "", using the default rules.
```
Comment (terrytangyuan) [inline pkg/apis/mxnet/v1beta1/types.go:52]: Describe the default behavior here

File: examples/train/byteps_dist_gpu_v1beta1.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+              command: ["bpslaunch"]
+              args: ["python3", "/usr/local/byteps/example/mxnet/train_imagenet_byteps.py", "--benchmark", "1", "--batch-size=32"]
+              volumeMounts:
+              - mountPath: /dev/shm
+                name: dshm
+              resources:
+                limits:
+                  nvidia.com/gpu: 8
+          volumes:
+          - name: dshm
```
Comment (terrytangyuan) [inline examples/train/byteps_dist_gpu_v1beta1.yaml:45]: Is this required?

**Comment by terrytangyuan** [comment]
Also please fix the CI build.

File: pkg/apis/mxnet/v1beta1/types.go (hunk 2 above)
Comment (jasonliu747) [reply pkg/apis/mxnet/v1beta1/types.go:48]: The default is actually `All` according to `defaults.go`. I believe it was a mistake, so drop by to fix it.

File: examples/train/byteps_dist_gpu_v1beta1.yaml (hunk 4 above)
Comment (jasonliu747) [reply examples/train/byteps_dist_gpu_v1beta1.yaml:45]: Yes, in order to launch BytePS successfully, you need `shm-size`

File: pkg/apis/mxnet/v1beta1/types.go (hunk 3 above)
Comment (jasonliu747) [reply pkg/apis/mxnet/v1beta1/types.go]: 👌

**Comment by jasonliu747** [comment]
> Also please fix the CI build.

Done, it's fixed.

**Pull request merged**